matching-notification-api/src/main/java/org/phenotips/matchingnotification/finder/internal/DefaultMatchFinderManager.java

```diff
@@ -66,75 +65,16 @@
     @Inject
     private PermissionsManager permissionsManager;
 
-
     @Override
-    public void findMatchesForAllPatients(Set<String> matchersToUse, boolean onlyCheckPatientsUpdatedAfterLastRun)
+    public void findMatchesForAllPatients(Set<String> serverIds, boolean onlyCheckPatientsUpdatedAfterLastRun)
     {
-        this.recordFindAllStart(matchersToUse);
-
         List<Patient> patients = this.getPatientsList();
 
-        for (Patient patient : patients) {
-            this.logger.debug("Finding matches for patient {}.", patient.getId());
-            this.findMatches(patient, matchersToUse, onlyCheckPatientsUpdatedAfterLastRun);
-        }
-
-        this.recordFindAllEnd(matchersToUse);
-    }
-
-
-    @Override
-    public List<PatientMatch> findMatches(Patient patient)
-    {
-        return this.findMatches(patient, null, false);
-    }
-
-
-    private List<PatientMatch> findMatches(Patient patient,
-            Set<String> matchersToUse, boolean onlyCheckPatientsUpdatedAfterLastRun)
-    {
-        List<PatientMatch> matches = new LinkedList<>();
-
-        for (MatchFinder service : this.matchFinderProvider.get()) {
-            try {
-                if (matchersToUse != null && !matchersToUse.contains(service.getName())) {
-                    continue;
-                }
-
-                List<PatientMatch> foundMatches = service.findMatches(patient, onlyCheckPatientsUpdatedAfterLastRun);
-                matches.addAll(foundMatches);
-
-                this.logger.debug("Found {} matches by {}", foundMatches.size(), service.getClass().getSimpleName());
-
-                for (PatientMatch match : foundMatches) {
-                    this.logger.debug(match.toString());
-                }
-
-            } catch (Exception ex) {
-                this.logger.error("Failed to invoke match finder [{}]", service.getName(), ex);
-            }
-        }
-
-        return matches;
-    }
-
-    private void recordFindAllStart(Set<String> matchersToUse)
-    {
-        for (MatchFinder service : this.matchFinderProvider.get()) {
-            if (matchersToUse != null && !matchersToUse.contains(service.getName())) {
-                continue;
-            }
-            service.recordStartMatchesSearch();
-        }
-    }
-
-    private void recordFindAllEnd(Set<String> matchersToUse)
-    {
-        for (MatchFinder service : this.matchFinderProvider.get()) {
-            if (matchersToUse != null && !matchersToUse.contains(service.getName())) {
+        for (MatchFinder matchFinder : this.matchFinderProvider.get()) {
+            if (matchFinder instanceof LocalMatchFinder && !serverIds.contains("local")) {
```

Comment:
This is a bad pattern (`instanceof`). Instead we should design a system to know which matcher(s) to ruun, but one that doe snot rely on class names. E.g. see my comments above and below. Not sure whats the best system yet, need to discuss

Reply:
moved a check to the `LocalMatchFinder`